src/backend/Repositories/OffersRepository.cs

```diff
@@ -53,5 +54,16 @@ public List<Offer> GetAll()
                 .ToList();
             return offers;
         }
+
+        public Offer UpdateOffer (Offer oldOffer, UpdateOfferDto newOffer)
+        {
+            _db.Entry(oldOffer).CurrentValues.SetValues(newOffer);
+            if(newOffer.FoodName != null) oldOffer.Food.Name = newOffer.FoodName;
+            if(newOffer.FoodUnit != null) oldOffer.Food.Unit = newOffer.FoodUnit;
```

Comment:
You could have avoided these lines of code if you had created a nested DTO which would have contained these values.